.github/bin/build-pt-matrix-from-impacted-connectors.py

```diff
@@ -0,0 +1,145 @@
+    logging.debug("excluded(%s, %s) returns %s", item, excludes, result)
+    return result
+
+
+def expand_matrix(matrix):
+    include = matrix.pop("include", [])
+    exclude = matrix.pop("exclude", [])
+
+    # for every key in the matrix dict, convert its values to tuples of key and value
+    tuples = [[(k, v) for v in vals] for k, vals in matrix.items()]
+    logging.debug("tuples: %s", tuples)
+    # then calculate the product of such lists of tuples
+    # and convert tuples back to a dict stored in a list
+    combinations = list(map(dict, itertools.product(*tuples)))
+    logging.debug("combinations: %s", combinations)
+    # filter out excludes and add includes as the last step
+    # so that excluded combinations can be explicitly added back
+    return [item for item in combinations if not excluded(item, exclude)] + include
+
+
+def load_available_features_for_config(config, suites):
+    cmd = ['testing/bin/ptl', 'suite', 'describe', '--suite', ",".join(suites), '--config', "config-" + config, '--format', 'JSON']
+    logging.debug("executing: %s", " ".join(cmd))
+    process = subprocess.run(cmd,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             universal_newlines=True)
+    logging.debug("ptl suite describe: %s", process)
+    if process.returncode != 0:
+        logging.error("ptl suite describe failed: %s", process)
+        return {}
+    for line in process.stdout.splitlines():
+        if line.startswith('{'):
+            logging.debug("Parsing JSON: %s", line)
+            ptl_output = json.loads(line)
+            logging.debug("Handling JSON object: %s", ptl_output)
+            config_features = {(config, suite.get("name")): set([connector for testRun in suite.get("testRuns", []) for connector in testRun["environment"].get("features", [])])
+                                 for suite in ptl_output.get("suites", [])}
+            logging.debug("config_features: %s", config_features)
+            return config_features
+    logging.error("ptl suite describe hasn't returned any JSON line: %s", process)
+    return {}
+
+
+def load_available_features(configToSuiteMap):
+    available_connectors = {}
+    for config, suites in configToSuiteMap.items():
+        available_connectors.update(
+            load_available_features_for_config(config, suites)
+        )
+    return available_connectors
+
+
+def tested_features(available_connectors, config, suite):
+    return available_connectors.get((config, suite), [])
+
+
+def build(matrix_file, impacted_file, output_file):
+    matrix = yaml.load(matrix_file, Loader=yaml.Loader)
+    impacted_features = list(filter(None, [line.rstrip() for line in impacted_file.readlines()]))
+    logging.info("Read matrix: %s", matrix)
+    logging.info("Read impacted_features: %s", impacted_features)
+    items = expand_matrix(matrix)
```

Comment:
In #11446 we agreed not to attempt to expand the matrix as this is error-prone. It should be possible to filter out modules in the matrix without expanding it. Also, check that PR on how tests can be incorporated into this script.